contracts/interfaces/IAssetPool.sol

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity <0.9.0;
+
+interface IAssetPool {
+    /// @notice Accepts the given amount of collateral token as a deposit and
+    ///         mints underwriter tokens representing pool's ownership.
+    /// @dev Before calling this function, collateral token needs to have the
+    ///      required amount accepted to transfer to the asset pool.
+    function deposit(uint256 amount) external;
```

Comment:
I would also add `initiateWithdrawal` and `completeWithdrawal` to this interface. Withdrawal will always be a two-step process and even if not, only deposit function is needed for the upgrade.

Reply:
Done